spec/lib/core_ext/string/string_normalise_whitespace_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+describe String do
+  describe "#normalise_whitespace" do
+    it "removes leading and trailing whitespace" do
+      expect("  hello  ".normalise_whitespace).to eq("hello")
+    end
+
+    it "replaces multiple spaces with a single space" do
+      expect("hello  world".normalise_whitespace).to eq("hello world")
+    end
+
+    it "handles tabs and newlines" do
+      expect("hello\t\nworld".normalise_whitespace).to eq("hello world")
+    end
+
+    it "returns nil if string is empty after normalization" do
+      expect("   ".normalise_whitespace).to be_nil
+    end
+
+    it "returns nil if string is empty" do
+      expect("".normalise_whitespace).to be_nil
+    end
+
+    context "with UTF-8 encoded strings" do
+      it "removes zero-width joiners (ZWJ)" do
+        string_with_zwj = "1234‍567890"
```

Comment:
I'd suggest using explicit `\u200D` characters here to ensure they're not missed. Certainly in my editor the characters aren't rendered so it just looks like the same string before and after.